File: 3243/3243.py
class Solution:
    def shortest_distance_after_road_addition_qu(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.shortest_distance_after_road_addition_qu(root.left), self.shortest_distance_after_road_addition_qu(root.right))

File: 3243/3243.go
func shortestDistanceAfterRoadAddit(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := shortestDistanceAfterRoadAddit(root.Left)
	r := shortestDistanceAfterRoadAddit(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}